### PR TITLE
[bazel] Convert most of silicon_creator to the new rules 

### PR DESCRIPTION
**Pull request by cfrantz**
This PR depends on #19650.  You need only review the last commit.

### CODE REVIEW

File: sw/device/silicon_creator/lib/sigverify/sphincsplus/test/BUILD [hunk 1]
```diff
@@ -154,10 +158,13 @@ autogen_cryptotest_header(
     tool = ":sphincsplus_set_testvectors",
 )
 
-opentitan_functest(
+opentitan_test(
     name = "verify_test_kat0",
     srcs = ["verify_test.c"],
-    targets = ["cw310_test_rom"],  # Test set is too large for Verilator or ROM
+    exec_env = {
+        # Test set is too large for Verilator or ROM
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/sigverify/sphincsplus/test/BUILD:165]: What do you mean by "too large for ROM"? Too large to run as a "test in ROM" type test?

**Review by timothytrippel — APPROVED** [review]

File: sw/device/silicon_creator/lib/BUILD [hunk 2]
```diff
@@ -396,24 +404,30 @@ cc_test(
     ],
 )
 
-opentitan_functest(
+opentitan_test(
     name = "crc32_functest",
     srcs = ["crc32_functest.c"],
+    exec_env = EARLGREY_TEST_ENVS,
+    verilator = verilator_params(
```
Comment (pamaury) [inline sw/device/silicon_creator/lib/BUILD:411]: This timeout was not there before, was it too small or did the default for verilator changed in your PR?

**Review by pamaury — APPROVED** [review]

File: sw/device/silicon_creator/lib/BUILD (hunk 2 above)
Comment (cfrantz) [reply sw/device/silicon_creator/lib/BUILD:411]: I experienced a timeout when I tested this rule.  I _also_ experience a timeout (on my workstation) on this test prior to the rule refactoring.

File: sw/device/silicon_creator/lib/sigverify/sphincsplus/test/BUILD (hunk 1 above)
Comment (cfrantz) [reply sw/device/silicon_creator/lib/sigverify/sphincsplus/test/BUILD:165]: I simply preserved these comments from the prior rule.

Comment (cfrantz) [reply sw/device/silicon_creator/lib/sigverify/sphincsplus/test/BUILD]: After running into an error with another test, I've determined what this comment is meant to convey.

Some of the crypto tests have large data sets that make the test binary too large to be a ROM_EXT stage executed by the real ROM (ie:  greater than 64K).  The test_rom doesn't enforce any of the usual limits, so these tests are fine to be executed in ROM_EXT position by the test_rom.

I've updated the comment to be more clear.

**Pull request merged**